### PR TITLE
feat: add taskType to find query(MAPCO-4570)

### PR DESCRIPTION
**Pull request by RonitKissis**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                       |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/DAL/repositories/jobRepository.ts [hunk 1]
```diff
@@ -80,30 +80,44 @@ export class JobRepository extends GeneralRepository<JobEntity> {
       domain: req.domain,
     };
 
-    if (req.fromDate != undefined && req.tillDate != undefined) {
+    if (req.fromDate !== undefined && req.tillDate != undefined) {
```
Comment (CL-SHLOMIKONCHA) [inline src/DAL/repositories/jobRepository.ts]: consider remove this 'undefined' condition

File: src/DAL/repositories/jobRepository.ts [hunk 2]
```diff
@@ -80,30 +80,44 @@ export class JobRepository extends GeneralRepository<JobEntity> {
...
+    } else if (req.tillDate !== undefined) {
       filter.updateTime = LessThanOrEqual(req.tillDate);
-    } else if (req.fromDate != undefined) {
+    } else if (req.fromDate !== undefined) {
       filter.updateTime = MoreThanOrEqual(req.fromDate);
     }
 
     for (const key of Object.keys(filter)) {
-      if (filter[key] == undefined) {
+      if (filter[key] === undefined) {
```
Comment (CL-SHLOMIKONCHA) [inline src/DAL/repositories/jobRepository.ts:92]: consider remove this 'undefined' condition

File: src/DAL/repositories/jobRepository.ts [hunk 3]
```diff
@@ -80,30 +80,44 @@ export class JobRepository extends GeneralRepository<JobEntity> {
...
-    const queryBuilder = this.createQueryBuilder().select('job').from(JobEntity, 'job').where(filter);
+    const queryBuilder = this.createQueryBuilder('job').select().where(filter);
     if ((req.types?.length ?? 0) > 0) {
       queryBuilder.andWhere('job.type IN (:...types)', { types: req.types });
     }
     if ((req.statuses?.length ?? 0) > 0) {
       queryBuilder.andWhere('job.status IN (:...statuses)', { statuses: req.statuses });
     }
+
+    if (req.taskType !== undefined) {
```
Comment (CL-SHLOMIKONCHA) [inline src/DAL/repositories/jobRepository.ts:107]: consider remove this 'undefined' condition

File: tests/integration/jobs/jobs.spec.ts [hunk 4]
```diff
@@ -621,6 +619,42 @@ describe('job', function () {
         expect(response).toSatisfyApiSpec();
       });
 
+      it('should get all jobs with specific task type and return 200', async function () {
```
Comment (CL-SHLOMIKONCHA) [inline tests/integration/jobs/jobs.spec.ts]:
```suggestion
      it('should get all jobs with task counts by task type and return 200', async function () {
```

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: src/DAL/repositories/jobRepository.ts (hunk 1 above)
Comment (RonitKissis) [reply src/DAL/repositories/jobRepository.ts]: cant, it is required bu the linter. if this condition is not set, the linter gives the following warning:  `warning  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly ` .
more over, this condition : `req.fromDate !== undefined` returns boolean while this req.fromDate returns undefined

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Review by netanelC — APPROVED** [review]

**Pull request merged**